packages/client/src/pages/MyTrecipes/MyTrecipes.tsx

```diff
@@ -1,60 +1,221 @@
 import React from 'react';
 import './MyTrecipes.scss';
-import { FilterButton } from './Filter/FilterButton';
+import { FilterButton, FilterButtonTypes } from './Filter/FilterButton';
 import { Button } from '../../components/Button/Button';
 import TrecipeCard from './TrecipeCard/TrecipeCard';
-import { FilterSelector } from './Filter/FilterSelector';
+import { FilterSelector, FilterSelectorTypes } from './Filter/FilterSelector';
 import { IconProp } from '@fortawesome/fontawesome-svg-core';
 import { connect } from 'react-redux';
 import { RootState } from '../../redux';
 import { fetchAllTrecipes } from '../../redux/TrecipeList/action';
 import { bindActionCreators, Dispatch } from 'redux';
 import { showModal } from '../../redux/Modal/action';
-import { Link, withRouter, RouteComponentProps } from 'react-router-dom';
+import { Link, RouteComponentProps, withRouter } from 'react-router-dom';
 import TrecipePopup, { TrecipePopupType } from '../../components/TrecipePopup/TrecipePopup';
 import Trecipe from '../../../../shared/models/trecipe';
 
 type MyTrecipesProps = ReturnType<typeof mapStateToProps> &
     ReturnType<typeof mapDispatchToProps> &
     RouteComponentProps;
 
-class MyTrecipes extends React.Component<MyTrecipesProps, {}> {
+interface FilterWithState {
+    filterName: string;
+    selected: boolean;
+}
+
+export interface MyTrecipesState {
+    filterStates: Array<FilterWithState>;
+    privacyState: FilterSelectorTypes;
+    filteredTrecipesWithProgress: Array<Trecipe>;
+    filteredTrecipesWithPrivacy: Array<Trecipe>;
+}
+
+class MyTrecipes extends React.Component<MyTrecipesProps, MyTrecipesState> {
     private static contextFilters: string[] = ['All Trecipes', 'Completed', 'In Progress', 'To Do'];
 
+    constructor(props: Readonly<MyTrecipesProps>) {
+        super(props);
+        let filterStates: Array<FilterWithState> = MyTrecipes.contextFilters.map((name) =>
+            name === FilterButtonTypes.ALL
+                ? { filterName: name, selected: true }
+                : { filterName: name, selected: false }
+        );
+        this.state = {
+            filteredTrecipesWithProgress: [],
+            filteredTrecipesWithPrivacy: [],
+            filterStates: filterStates,
+            privacyState: FilterSelectorTypes.ANY,
+        };
+    }
+
     componentDidMount(): void {
         this.props.fetchAllTrecipes();
     }
 
+    componentDidUpdate(
+        prevProps: Readonly<MyTrecipesProps>,
+        prevState: Readonly<MyTrecipesState>,
+        snapshot?: any
```

Comment:
These can be removed if we are not using them

Reply:
Removed this function as a whole 🤦‍♂️